Configured profile tomcat.pp and module params.pp

bdeputat opened the pull request.

bdeputat committed:
Configured profile tomcat.pp and module params.pp

bdeputat commented:
Good day to all. I ask you to see the changes in the manifests ::profile::tomcat.pp and ::tomcat::params.pp  and say your wishes. Thank you!

ipleten commented on modules/profile/manifests/tomcat.pp:
```diff
@@ -1,26 +1,32 @@
-) {
-
+class profile::tomcat(
+  String $man_user       = 'manager',  
```
Why we have variable type specified here?
But in other place no String List or Integers?
It should be consistent across all classes.

bdeputat replied:
Error corrected.

ipleten commented on modules/tomcat/manifests/params.pp:
```diff
@@ -2,29 +2,31 @@
+$password             = "$profile::tomcat::password", 
+$host_name            = "$profile::tomcat::host_name",
+$docBase              = "$profile::tomcat::docBase", 
+){
```
Why some variables are parameters...

bdeputat replied:
Thank you for your comments. Below explanation ...

ipleten commented on modules/tomcat/manifests/params.pp:
```diff
@@ -2,29 +2,31 @@
-$catalina_base    = '/var/lib/tomcat/webapps' 
-$confdir          = '/etc/tomcat'
-$packages         = ['tomcat', 'tomcat-webapps', 'tomcat-admin-webapps']
+$user 				  = 'tomcat'
```
but other are hard-coded in module code?
Move them to parameters section as well.

bdeputat replied:
Variables like $host_name are global and are transmitted from the profile tomcat.pp . Variables like such as $ user = 'tomcat', $ group = 'tomcat', $catalina_home        = '/usr/share/tomcat' - these are default local parameters that do not need to be parameterized.

bdeputat committed:
Fixed bugs. Modified profile tomcat.pp and module tomcat

ipleten commented on modules/tomcat/manifests/params.pp:
```diff
@@ -3,24 +3,26 @@
-# config variables for aplication
+# config global variables for aplication
 $man_user		      = "$profile::tomcat::man_user",	
 $password             = "$profile::tomcat::password", 
```
oh my.... Why do you  have variables from PROFILE like default in MODULE?!

savroman committed:
add parameters to maven profile (#2)

* add parameters to maven profile

* make fixes according to review comments

* fixed review comments except yumrepo

turnopil committed:
Mysql (#5)

* Minor fix in mysql nodes

* Added parametrized class Mysql

* Minor fixes for root pass & install

* Fixed template & profiles

* Fixed var & delete empty class

* rm var duplication

* fixed var name

* Add some comments for variables. Profiles now not parametrized

* Var fixed

* Removed comas in mysql profiles

* Move mysql ver to install section & added default values to db defined type

* Remove indentationin init manifest

* yumrepo added in profiles

* yumrepo mysql  moved to profiles

* Made two spaces indentation on all of manifests && define Yumrepo only in profiles

* Fixed vars in class mysql

* Final version of yumrepo

bdeputat committed:
Fixed bugs in the module and profile tomcat.pp. The default settings for example application example.local

bdeputat committed:
Fixed bugs in the module and profile tomcat.pp. The default settings for example application example.local

bdeputat committed:
Fixed bugs in the module and profile tomcat.pp. The default settings for example application example.local

bdeputat commented:
Good evening all.
Please check the new version of the manifest for the server tomcat
Fixed bugs in the module and profile. The default settings for example.local

turnopil reviewed (commented):
+  if ($docBase == 'sample') 
+  {
+	$appdir             = "$catalina_home/webapps/$docBase"
+  }
+  else {
+  	$appdir           = "$catalina_home/webapps/$docBase"

Maybe better left only $appdir="$catalina_home/webapps/$docBase" . IF and ELSE do the same things in your code

ipleten commented on modules/profile/manifests/tomcat.pp:
```diff
@@ -1,36 +1,54 @@
+  
+  class { 'tomcat':
+    tomcat_version    	=> $tomcat_version,
+    dns_name      		=> $dns_name,
```
indentation.

bdeputat committed:
Fixed bugs in the If els

bdeputat commented:
turnopil thanks for the comment.
I agree.
Already fixed)

ipleten commented on modules/tomcat/manifests/init.pp:
```diff
@@ -5,26 +5,111 @@
+  $users_conf_file    = "$confdir/tomcat-users.xml"
+
+# # Making sure the  aplication directory is present/created Create
+  if ($docBase == 'sample') 
```
I don't understand why you need that if statement?

ipleten commented on modules/tomcat/manifests/init.pp:
```diff
@@ -5,26 +5,111 @@
+     require          => File["$users_conf_file"],
   }
+
   service { 'httpd':
```
You can't manage http service in tomcat module.

ipleten commented on modules/tomcat/manifests/proxy.pp:
```diff
@@ -5,30 +5,32 @@
 )
 {
+  # config variables for mod_proxy
+  $proxy_conf_file      = "/etc/httpd/conf.d/tomcat.conf"
```
Also you can manage httpd from tomcat module.
You need to have defined type in httpd module which you will call in  profiles::webapp::tomcat profiles.

ipleten commented on modules/tomcat/templates/proxy.conf.epp:
```diff
@@ -1,10 +1,10 @@
-<%- | String $host_name
+<%- | String $dns_name
```
All  template should be moved to httpd module

bdeputat replied:
moved to web profile

ipleten commented on modules/tomcat/templates/tomcat.conf.epp:
```diff
@@ -36,7 +36,7 @@ CATALINA_TMPDIR="/var/cache/tomcat/temp"
 # You can pass some parameters to java here if you wish to
 #
-JAVA_OPTS="-Xmin<%= $java_ram %> -Xmax<%= $java_ram %>"
+JAVA_OPTS="-Xmin<%= $java_heap %> -Xmax<%= $java_heap %>"
```
Are you sure  that java heap memory setting are called Xmin and Xmax but not Xms and Xmx?

ipleten reviewed (changes requested):
You can't manage anything related to httpd in tomcat module.

oleksdiam committed:
Syslog (#6)

* sonar node made through profile

* using firewall module realized

* modules/profile/manifests/tomcat.pp

* rsyslog sends remote messages from apps with queues

* add rsyslog client to profile

* get back jenkins & tomcat profiles, insert logging into roles

* unused code is throwed out

* unused code is thrown out

* rsyslog::server has been rebuilt as a class

* rsyslog::server has been rebuilt as a class

* rsyslog::server has been rebuilt as a class,node sonar works with logging

* additional template for rsyslog

* added logging postgres to syslog

* IP of rsyslog server changed to ...15

* some little makeup

* ssh user oved to basenode

* rsyslog module was rebuilt

* rsyslog module was rebuilt

* little fricking bug

* little final fricking bug. Tested

bdeputat committed:
The role webapp and profile tomсat has been modified. Excluded httpd from the module tomcat

bdeputat committed:
The role webapp and profile tomсat has been modified. Excluded httpd from the module tomcat

ipleten commented on modules/profile/manifests/tomcat/tomcat.pp:
```diff
@@ -0,0 +1,59 @@
+class profile::tomcat::tomcat
```
profile::tomcat::tomcat? really? 
Why not just profiles::tomcat?

bdeputat replied:
fixed

ipleten commented on modules/profile/manifests/tomcat/tomcat.pp:
```diff
@@ -0,0 +1,59 @@
+     dns_name         => $dns_name,
+  }
+
+  base::ssh_user { $ssh_user:
```
in basenode?

ipleten commented on modules/profile/manifests/tomcat/proxy.pp:
```diff
@@ -0,0 +1,33 @@
+    'dns_name'        => $dns_name,
+  }
+
+  file { $proxy_conf_file:
```
that would be great to move to http MODULE, but i'm fine with such solution as well.

bdeputat replied:
httpd module will use for other people..

bdeputat committed:
Fixed site.pp include role::webapp. Fixed manifest profile::basenode and profile::webapp. The module rsyslog::client is included in the profiles profile::webapp::tomcat and profile::webapp::proxy

bdeputat committed:
Fixed site.pp include role::webapp. Fixed manifest profile::basenode and profile::webapp. The module rsyslog::client is included in the profiles profile::webapp::tomcat and profile::webapp::proxy

bdeputat committed:
Update webapp.pp

bdeputat committed:
Update init.pp

Fixed site.pp include role::webapp. Fixed manifest profile::basenode and profile::webapp. The module rsyslog::client is included in the profiles profile::webapp::tomcat and profile::webapp::proxy

ipleten commented on modules/mysql/manifests/install.pp:
```diff
@@ -6,22 +6,11 @@
 package { 'mysql-community-server':
   ensure  => present,
-  require => Package['mysql-community-repo'],
+  require => Yumrepo['mysql-repo'],
```
?
